File: Semester 4/Advanced Programming/RpcProtocolC#/Services/IServices.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Domain;
namespace services
{
    public interface IServices
    {
        void login(User user, IObserver client);
        void logout(IObserver obs);
        List<Cursa> getAllCurse();
        Cursa findByDestinationAndDate(string destination, DateTime date);
        List<Rezervare> getAllBookings(int id);
        void makeBooking(List<Rezervare> rezervari);
    }
}

File: Semester 4/Advanced Programming/RmiHibernate/ApplicationServices/src/main/java/app/services/IServices.java
package app.services;

import app.model.Cursa;
import app.model.Rezervare;
import app.model.User;

import java.time.LocalDateTime;
import java.util.List;

public interface IServices{

    void login(User user, IObserver client) throws ServiceException;
    void logout(IObserver obs) throws ServiceException;
    List<Cursa> getAllCurse() throws ServiceException;

    Cursa findByDestinationAndDate(String destination, LocalDateTime date) throws ServiceException;

    List<Rezervare> getAllBookings(Integer id) throws ServiceException;

    void makeBooking(List<Rezervare> rezervari) throws ServiceException;
}
